### PR TITLE
xl: Rename Force to Immediate for clarity

### PR DESCRIPTION
**Pull request by vadmeste**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Comment by klauspost** [comment]
`2023-11-28T03:13:27.5234227Z ##[error]cmd\xl-storage_windows_test.go:67:5: unknown field Force in struct literal of type DeleteOptions`

File: cmd/storage-datatypes.go [hunk 1]
```diff
@@ -23,8 +23,8 @@ import (
 
 // DeleteOptions represents the disk level delete options available for the APIs
 type DeleteOptions struct {
-	Recursive bool `msg:"r"`
-	Force     bool `msg:"f"`
+	Recursive      bool `msg:"r"`
+	ImmediatePurge bool `msg:"ip"`
```
Comment (klauspost) [inline cmd/storage-datatypes.go]: If it isn't too much of a hazzle, I think "Immediate" is enough. "Purge" is implied by it being a delete function.
```suggestion
	Immediate bool `msg:"i"`
```

**Review by klauspost — APPROVED** [review]
I can live with the long form, so approving, but I'd prefer a shorter name since it doesn't give additional information.

File: cmd/xl-storage.go [hunk 2]
```diff
@@ -2157,7 +2158,7 @@ func (s *xlStorage) CheckParts(ctx context.Context, volume string, path string,
 // move up the tree, deleting empty parent directories until it finds one
 // with files in it. Returns nil for a non-empty directory even when
 // recursive is set to false.
-func (s *xlStorage) deleteFile(basePath, deletePath string, recursive, force bool) error {
+func (s *xlStorage) deleteFile(basePath, deletePath string, recursive, immediatePurge bool) error {
```
Comment (harshavardhana) [inline cmd/xl-storage.go]:
```suggestion
func (s *xlStorage) deleteFile(basePath, deletePath string, recursive, immediate bool) error {
```

File: cmd/xl-storage.go [hunk 3]
```diff
@@ -2170,7 +2171,7 @@ func (s *xlStorage) deleteFile(basePath, deletePath string, recursive, force boo
 
 	var err error
 	if recursive {
-		err = s.moveToTrash(deletePath, true, force)
+		err = s.moveToTrash(deletePath, true, immediatePurge)
```
Comment (harshavardhana) [inline cmd/xl-storage.go]:
```suggestion
		err = s.moveToTrash(deletePath, true, immediate)
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**